### PR TITLE
Bump MSTest

### PR DESCRIPTION
**Pull request by pauld-msft**
newest MSTest packages

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -21,10 +21,10 @@
     <PackageVersion Include="MinVer" Version="5.0.0" />
     <PackageVersion Include="Moq" Version="4.18.4" />
     <PackageVersion Include="morelinq" Version="4.2.0" />
-    <PackageVersion Include="MSTest.TestFramework" Version="3.5.1" />
-    <PackageVersion Include="MSTest.Analyzers" Version="3.5.1" />
-    <PackageVersion Include="MSTest.TestAdapter" Version="3.5.1" />
-    <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.10.0" />
+    <PackageVersion Include="MSTest.TestFramework" Version="3.5.2" />
```
Comment (Evangelink) [inline Directory.Packages.props:24]: You probably need to also add `MSTest` metapackage (it is used here https://github.com/microsoft/component-detection/blob/main/test/Microsoft.ComponentDetection.VerificationTests/Microsoft.ComponentDetection.VerificationTests.csproj#L21)

```suggestion
    <PackageVersion Include="MSTest" Version="3.5.2" />
    <PackageVersion Include="MSTest.TestFramework" Version="3.5.2" />
```

Comment (pauld-msft) [reply Directory.Packages.props:24]: good find, removing that ref in favor of `MSTest.TestAdapter` and `Microsoft.NET.Test.Sdk`, similar to other test packages that are not packaged

File: test/Microsoft.ComponentDetection.VerificationTests/Microsoft.ComponentDetection.VerificationTests.csproj [hunk 2]
```diff
@@ -18,7 +18,8 @@
         IF TESTING LOCALLY, UNCOMMENT ABOVE AND COMMENT BELOW -->
         <PackageReference Include="FluentAssertions" />
         <PackageReference Include="Microsoft.Extensions.Logging" />
-        <PackageReference Include="MSTest" />
+        <PackageReference Include="MSTest.TestAdapter "/>
```
Comment (Evangelink) [inline test/Microsoft.ComponentDetection.VerificationTests/Microsoft.ComponentDetection.VerificationTests.csproj:21]: This is missing some packages:

```suggestion
        <PackageReference Include="MSTest.Analyzers"/>
        <PackageReference Include="MSTest.TestAdapter "/>
        <PackageReference Include="MSTest.TestFramework"/>
```

**Review by Evangelink — APPROVED** [review]

File: test/Microsoft.ComponentDetection.VerificationTests/Microsoft.ComponentDetection.VerificationTests.csproj (hunk 2 above)
Comment (pauld-msft) [reply test/Microsoft.ComponentDetection.VerificationTests/Microsoft.ComponentDetection.VerificationTests.csproj:21]: we get that package and `MSTest.TestFramework` from https://github.com/microsoft/component-detection/blob/main/test/Directory.Build.props#L6-L7 , removing `MSTest.TestFramework` from this csproj

**Comment by FernandoRojo** [comment]
Closed as these have already been updated

**Pull request closed without merging**